### PR TITLE
Drawing subtask into ES6

### PR DESCRIPTION
**Pull request by srallen**
Staging branch URL: https://drawing-subtask-es6.pfe-preview.zooniverse.org/

This refactors the drawing subtask into its own ES6 component in preparation for #4544. Also refactors the drawing tool root component into ES6

# Required Manual Testing

- [x] Does the non-logged in home page render correctly?
- [x] Does the logged in home page render correctly?
- [x] Does the projects page render correctly?
- [x] Can you load project home pages?
- [ ] Can you load the classification page?
- [ ] Can you submit a classification?
- [x] Does talk load correctly?
- [x] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [x] Does it work on mobile?
- [x] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [x] Are the tests passing locally and on Travis?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

File: app/classifier/drawing-tools/components/DetailsSubTaskForm/DetailsSubTaskForm.jsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+  componentDidUpdate() {
+    // What is this even doing? When this was in the root component, it was always being called for any update
+    // Only thing changing is the transform prop. 
+    // I've passed all of the parent's props to this because I don't know what this was reacting to
+    // In the modal-form code, this method is for repositioning the underlay's width and height
+    // But the opacity is 0, so we can't see it!
+    // The class for the StickyForm is updating the styles for the pointer and form position
+    // but the behavior seems the same if I comment out the following line:
+    if (this.detailsForm) this.detailsForm.reposition();
```
Comment (eatyourgreens) [inline app/classifier/drawing-tools/components/DetailsSubTaskForm/DetailsSubTaskForm.jsx]: @simensta might know the answer to this. My best guess is it keeps details on the screen after a task annotation has changed eg. after typing text into a text box.

File: app/classifier/drawing-tools/components/DetailsSubTaskForm/DetailsSubTaskForm.jsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    this.props.onFormClose();
+  }
+  render() {
+    const { tasks, toolProps } = this.props;
+
+    const detailsAreComplete = this.areDetailsComplete(tasks, toolProps);
+
+    return (
+      <StickyModalForm
+        ref={(node) => { this.detailsForm = node; }}
```
Comment (eatyourgreens) [inline app/classifier/drawing-tools/components/DetailsSubTaskForm/DetailsSubTaskForm.jsx]: Fun fact I recently learned while debugging weird ref behaviour: React will call this callback with `null` before updating a ref, leading to situations where `this.detailsForm` might be null.

File: app/classifier/tasks/components/TaskInputField/TaskInputField.jsx [hunk 3]
```diff
@@ -132,7 +132,7 @@ export class TaskInputField extends React.Component {
   }
 
   unFocus() {
-    this.field.dataset.focus = false;
+    if (this.field) this.field.dataset.focus = false;
```
[the remark this answers is not in the document]
Comment (srallen) [reply app/classifier/tasks/components/TaskInputField/TaskInputField.jsx]: I'm going to pull this out into its own PR and add a few tests

[the remark this answers is not in the document]
Comment (eatyourgreens) [reply app/classifier/tasks/components/TaskInputField/TaskInputField.jsx]: Sounds good to me. I'd rather get this fix out without waiting to review this PR.

[the remark this answers is not in the document]
Comment (srallen) [reply app/classifier/tasks/components/TaskInputField/TaskInputField.jsx]: Ok, this doesn't entirely fix the issue. There's another focus bug going on in Edge with the drawing task inputs. I'm debugging it now... hope to have something up after lunch.

**Comment by eatyourgreens** [comment]
On DCW, the subtask text box expands until it's taller than the browser viewport. That might be an existing bug with the text task.

To trigger it, enter some text, close the subtask modal then click on the line to open it again.

Screenshot showing that the text task input has expanded until it is taller than the window.

**Comment by eatyourgreens** [comment]
I've tried this out on Penguin Watch and Decoding the Civil War without any problems, except for the text box sizing noted above.

**Comment by simensta** [comment]
The text box sizing bug is an existing issue, see #4095.

**Comment by srallen** [comment]
@eatyourgreens or @simensta is this ok to merge? If so, it could tomorrow after SGL?

**Comment by eatyourgreens** [comment]
Sounds good to me. I didn’t see any problems when I tried it out.

**Pull request merged**